### PR TITLE
Add configurable batch size

### PR DESCRIPTION
**Pull request by noodlefrenzy**
I found it far easier to used my "cooked" tfrecords format as an input source than cope with image reading during training. Also, found it much easier to just deal with both inputs and sampling in tensor space than in "real"(numpy/scipy)-space. Also, swapped your ImagePool implementation for my ImageCache implementation since it starts using cached images a lot more aggressively.

Dropping your "join-both-images-into-a-single-array" logic as it makes scoring more tedious and is just kind of weird.

I also dropped your "early-out" logic flag in favor of the more tensorflow-thonic coordinator-based quitting, and do the same on exceptions, _and_ save the model checkpoint in a finally block so it should always be saved.

One note: I have _not_ tested the model with the cycleganTest.py script, I'll do that tomorrow morning.

### CODE REVIEW

File: cyclegan.py [hunk 1]
```diff
@@ -3,23 +3,19 @@
...
 import numpy as np
 import scipy.misc
 import argparse
 import tensorflow as tf
 import utils
 
-#ensure reproducability
+# ensure reproducability
 random_seed = 1234
 np.random.seed(random_seed)
```
Comment (olgaliak) [inline cyclegan.py]: you mentioned you wanted to get rid of seed setting? Maybe make a  optional cmdline switch to pass it seed?

File: cyclegan.py [hunk 2]
```diff
@@ -29,315 +25,303 @@
...
+def to_image(data):
+    return tf.image.convert_image_dtype((data + 1.) / 2., tf.uint8)
+
+
+def batch_to_image(batch):
+    return tf.map_fn(to_image, batch, dtype=tf.uint8)
+
+
+class Images():
+    def __init__(self, tfrecords_file, image_size=256, batch_size=1, num_threads=2, shuffle=True, name=''):
```
Comment (olgaliak) [inline cyclegan.py]: move all utils\Image related stuff to separated files\classes?
